VSDiagnostics/VSDiagnostics/VSDiagnostics/Diagnostics/Arithmetic/DivideIntegerByInteger/DivideIntegerByIntegerAnalyzer.cs

```diff
@@ -0,0 +1,60 @@
+﻿using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using VSDiagnostics.Utilities;
+
+namespace VSDiagnostics.Diagnostics.Arithmetic.DivideIntegerByInteger
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public class DivideIntegerByIntegerAnalyzer : DiagnosticAnalyzer
+    {
+        private static readonly SpecialType[] IntegerTypes =
+        {
+            SpecialType.System_Byte, SpecialType.System_Int16, SpecialType.System_Int32, SpecialType.System_Int64,
+            SpecialType.System_SByte, SpecialType.System_UInt16, SpecialType.System_UInt32, SpecialType.System_UInt64
+        };
+
+        private const DiagnosticSeverity Severity = DiagnosticSeverity.Warning;
+
+        private static readonly string Category = VSDiagnosticsResources.ArithmeticCategory;
+        private static readonly string Message = VSDiagnosticsResources.DivideIntegerByIntegerAnalyzerMessage;
+        private static readonly string Title = VSDiagnosticsResources.DivideIntegerByIntegerAnalyzerTitle;
+
+        internal static DiagnosticDescriptor Rule
+            =>
+                new DiagnosticDescriptor(DiagnosticId.DivideIntegerByInteger, Title, Message, Category, Severity,
+                    isEnabledByDefault: true);
```

Comment:
Oh nevermind. It's not an optional argument -- it's just a named argument. I did this here (and everywhere else) because I don't want a boolean argument since it doesn't say anything about what it does.